variables.tf

```diff
@@ -0,0 +1,188 @@
+variable "ipam" {
+  description = "Object of inputs for IPAM resources"
+  type = object({
+    description = optional(string)
+    region_name = optional(string)
+    tags        = optional(map(string))
+    cascade     = optional(bool)
+  })
+  default  = {}
+  nullable = false
+
+  validation {
+    condition     = try(var.ipam.description, null) != null
```

Comment:
This pattern feels a little awkward. Instead, just remove the "optional" marker to make description required. E.g.

```
    description = string
```

Remove this pattern everywhere for making an attribute required, and just use the builtin features of the type instead.